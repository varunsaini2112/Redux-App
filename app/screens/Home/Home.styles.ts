import { StyleSheet } from "react-native";
import { Padding } from "../../config/appStyles";

const styles = StyleSheet.create({
	mainContainer: {
		flex: 1,
		padding: Padding.m
	},
	textContainer: {
		padding: Padding.m
	}
});

export default styles;
