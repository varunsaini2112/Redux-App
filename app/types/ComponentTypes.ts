import { RootStackParamList } from ".";

type LinkType = {
	value: string;
	href: keyof RootStackParamList;
};

type LinkProps = {
	value: string;
	onPress: () => void;
};

export type { LinkType, LinkProps };
