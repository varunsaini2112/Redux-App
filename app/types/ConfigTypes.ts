type RootStackParamList = {
	Home: undefined;
	Count: undefined;
	Login: undefined;
	ToDo: undefined;
};

export type { RootStackParamList };
