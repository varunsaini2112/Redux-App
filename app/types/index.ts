export * from "./ConfigTypes";

export * from "./ComponentTypes";
