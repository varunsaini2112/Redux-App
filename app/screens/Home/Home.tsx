import { FC } from "react";
import { View } from "react-native";
import { StackScreenProps } from "@react-navigation/stack";

import styles from "./Home.styles";
import { H2, Link } from "../../components";
import { LinkType, RootStackParamList } from "../../types";

const links: LinkType[] = [
	{ value: "Count Program", href: "Count" },
	{ value: "To-Do List", href: "ToDo" }
];

const Home: FC<StackScreenProps<RootStackParamList, "Home">> = (props) => {
	const { navigation } = props;

	const onPressLink = (screenName: keyof RootStackParamList) => {
		navigation.navigate(screenName);
	};

	return (
		<View style={styles.mainContainer}>
			<View style={styles.textContainer}>
				<H2>Projects</H2>
			</View>
			<View style={styles.textContainer}>
				{links.map((link) => {
					const onPress = () => onPressLink(link.href);
					return <Link {...link} key={link.value} onPress={onPress} />;
				})}
			</View>
		</View>
	);
};

export default Home;
