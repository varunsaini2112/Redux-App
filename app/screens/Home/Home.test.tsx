import { fireEvent, render, screen } from "@testing-library/react-native";
import { RouteProp } from "@react-navigation/native";
import { StackNavigationProp } from "@react-navigation/stack";

import Home from "./Home";
import { RootStackParamList } from "../../types";

const navigation = {
	navigate: () => {}
} as StackNavigationProp<RootStackParamList, "Home">;
const route = {} as RouteProp<RootStackParamList, "Home">;

describe("Home", () => {
	it("should match snapshots", () => {
		const snapshot = render(<Home navigation={navigation} route={route} />);

		expect(snapshot).toMatchSnapshot();
	});

	it("should render projects links", () => {
		const spy = jest.spyOn(navigation, "navigate");
		render(<Home navigation={navigation} route={route} />);

		const toDoLink = screen.getByText("Count Program");
		fireEvent.press(toDoLink);

		expect(spy).toHaveBeenCalledWith("Count");
	});
});
