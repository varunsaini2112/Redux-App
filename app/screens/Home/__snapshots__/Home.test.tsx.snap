// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshots 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 12,
    }
  }
>
  <View
    style={
      {
        "padding": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#132631",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    >
      Projects
    </Text>
  </View>
  <View
    style={
      {
        "padding": 12,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          [
            {
              "color": "#999999",
              "fontSize": 14,
              "fontWeight": "500",
            },
            {
              "fontSize": 20,
            },
          ]
        }
      >
        Count Program
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          [
            {
              "color": "#999999",
              "fontSize": 14,
              "fontWeight": "500",
            },
            {
              "fontSize": 20,
            },
          ]
        }
      >
        To-Do List
      </Text>
    </View>
  </View>
</View>
`;
